dbms/src/IO/WriteBufferFromHDFS.cpp

```diff
@@ -32,9 +33,12 @@ struct WriteBufferFromHDFS::WriteBufferFromHDFSImpl
     {
         const size_t begin_of_path = hdfs_uri.find('/', hdfs_uri.find("//") + 2);
         const std::string path = hdfs_uri.substr(begin_of_path);
-        if (path.find("*?{") != std::string::npos)
+        if (path.find("^{*") != std::string::npos)
```

Comment:
What is the reason for this change?

Reply:
And it looks totally incorrect because it will search for substring.
Do we have a test for this exception?